testing/endpoint.go

```diff
@@ -564,6 +564,24 @@ func (endpoint *Endpoint) TimeoutOnClose(packet channeltypes.Packet) error {
 	return endpoint.Chain.sendMsgs(timeoutOnCloseMsg)
 }
 
+// QueryChannelUpgradeProof returns all the proofs necessary to execute UpgradeTry/UpgradeAck/UpgradeOpen.
+// It returns the proof for the channel on the counterparty chain, the proof for the upgrade attempt on the
+// counterparty chain, and the height at which the proof will succeed.
```

Comment:
```suggestion
// counterparty chain, and the height at which the proof was queried.
```

Reply:
I swear I saw this someplace and will succeed sounded too fancy not to add but can't find out where I saw it from now.

Reply:
😂 maybe here? https://github.com/cosmos/ibc-go/blob/d751bc91a5fb2ec26f482ed7dfd5099a33ebb019/testing/chain.go#L199 

but that godoc saying `succeeds` seems correct bc it's `height-1`

Reply:
that definitely looks like the inspiring docstring :smile: